src/test/java/org/fogbowcloud/manager/requests/api/local/http/ImageRequestControllerTest.java

```diff
@@ -0,0 +1,164 @@
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(SpringRunner.class)
+@WebMvcTest(value = ImageRequestController.class, secure = false)
+@PrepareForTest(ApplicationFacade.class)
+public class ImageRequestControllerTest {
+
+    private final String IMAGE_ENDPOINT = "/" + ImageRequestController.IMAGE_ENDPOINT;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private ApplicationFacade facade;
+
+    @Before
+    public void setUp() throws FogbowManagerException {
+        this.facade = spy(ApplicationFacade.class);
+        PowerMockito.mockStatic(ApplicationFacade.class);
+        given(ApplicationFacade.getInstance()).willReturn(this.facade);
+    }
+
+    // test case: Test getAllImages() when no images were found, it must return an empty JSON.
+    @Test
+    public void testGetAllImagesWhenHasNoData() throws Exception {
+        // set up
+        Map<String, String> imagesMap = new HashMap<>();
+        doReturn(imagesMap).when(this.facade).getAllImages(anyString(), anyString());
+
+        RequestBuilder requestBuilder = createRequestBuilder(IMAGE_ENDPOINT, getHttpHeaders(), "");
+
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        assertEquals("{}", result.getResponse().getContentAsString());
+        assertEquals(expectedStatus, result.getResponse().getStatus());
+    }
+
+    // test case: Test getAllImages() when there are some images.
+    @Test
+    public void testGetAllImagesWhenHasData() throws Exception {
+        // set up
+        Map<String, String> imagesMap = new HashMap<>();
+        imagesMap.put("image-id1", "image-name1");
+        imagesMap.put("image-id2", "image-name2");
+        imagesMap.put("image-id3", "image-name3");
+
+        doReturn(imagesMap).when(this.facade).getAllImages(anyString(), anyString());
+
+        RequestBuilder requestBuilder = createRequestBuilder(IMAGE_ENDPOINT, getHttpHeaders(), "");
+
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        assertEquals(expectedStatus, result.getResponse().getStatus());
+
+        TypeToken<Map<String, String>> token = new TypeToken<Map<String, String>>(){};
+        Map<String, String> resultMap = new Gson().fromJson(result.getResponse().getContentAsString(), token.getType());
+        assertTrue(resultMap.size() == 3);
```

Comment:
assertEquals fits better. If there are more calls like that, please, change them.

Reply:
Done!